cachecontrol/controller.py

```diff
@@ -277,9 +277,11 @@ def cached_request(self, request: PreparedRequest) -> HTTPResponse | Literal[Fal
 
     def conditional_headers(self, request: PreparedRequest) -> dict[str, str]:
         resp = self._load_from_cache(request)
-        new_headers = {}
+        if not resp:
+            return {}
+        with resp:
```

Comment:
I'm scratching my head why unraisablehook isn't detecting this missing with. Perhaps requests is closing the connection on GC without issuing a ResourceWarning?

Reply:
It's also possible a SeparateBodyCache isn't used in the tests (I haven't checked)?

Reply:
I think `test_body_actually_stored_separately` is doesn't test the case where theres an outdated entry in cache, which is when a ResourceWarning should appear.